Mathlib/Order/UpperLower/Basic.lean

```diff
@@ -1028,19 +1028,27 @@ instance UpperSet.isTotal_le : IsTotal (UpperSet α) (· ≤ ·) := ⟨fun s t =
 instance LowerSet.isTotal_le : IsTotal (LowerSet α) (· ≤ ·) := ⟨fun s t => s.lower.total t.lower⟩
 #align lower_set.is_total_le LowerSet.isTotal_le
 
-noncomputable instance : CompleteLinearOrder (UpperSet α) :=
-  { UpperSet.completelyDistribLattice with
-    le_total := IsTotal.total
-    decidableLE := Classical.decRel _
-    decidableEq := Classical.decRel _
-    decidableLT := Classical.decRel _ }
-
-noncomputable instance : CompleteLinearOrder (LowerSet α) :=
-  { LowerSet.completelyDistribLattice with
-    le_total := IsTotal.total
-    decidableLE := Classical.decRel _
-    decidableEq := Classical.decRel _
-    decidableLT := Classical.decRel _ }
+noncomputable instance UpperSet.instLinearOrder : LinearOrder (UpperSet α) := by
+  classical exact Lattice.toLinearOrder _
+
+noncomputable instance LowerSet.instLinearOrder : LinearOrder (LowerSet α) := by
+  classical exact Lattice.toLinearOrder _
+
+noncomputable instance UpperSet.instCompleteLinearOrder : CompleteLinearOrder (UpperSet α) where
+  __ := completelyDistribLattice
+  __ := LinearOrder.toBiheytingAlgebra
+  le_total := IsTotal.total
+  decidableLE := Classical.decRel _
+  decidableEq := Classical.decRel _
+  decidableLT := Classical.decRel _
```

Comment:
I worry this is going to create a diamond with the `instLinearOrder` instance above. Can't you use `__ := instLinearOrder` here?

Reply:
Both are definitely `Classical.decRel _` because they prove decidability of inclusions of sets.

Spread notation seems to be broken/uncover a diamond. This works:
```
noncomputable instance UpperSet.instCompleteLinearOrder : CompleteLinearOrder (UpperSet α) where
  __ := completelyDistribLattice
  __ := instLinearOrder
  himp := LinearOrder.toBiheytingAlgebra.himp
  le_himp_iff := LinearOrder.toBiheytingAlgebra.le_himp_iff
  compl := LinearOrder.toBiheytingAlgebra.compl
  himp_bot := LinearOrder.toBiheytingAlgebra.himp_bot
  sdiff := LinearOrder.toBiheytingAlgebra.sdiff
  hnot := LinearOrder.toBiheytingAlgebra.hnot
  sdiff_le_iff := LinearOrder.toBiheytingAlgebra.sdiff_le_iff
  top_sdiff := LinearOrder.toBiheytingAlgebra.top_sdiff
```
This works too:
```
noncomputable instance UpperSet.instCompleteLinearOrder : CompleteLinearOrder (UpperSet α) :=
  { completelyDistribLattice, instLinearOrder, LinearOrder.toBiheytingAlgebra with }
```
This doesn't:
```
noncomputable instance UpperSet.instCompleteLinearOrder : CompleteLinearOrder (UpperSet α) where
  __ := completelyDistribLattice
  __ := instLinearOrder
```

Reply:
Is it possible to have "tests" to confirm the definitional equality (`with_reducible_and_instances rfl`?) of diamonds?

Reply:
How would that help here? I think it's a metavariable issue with the spread notation, not an actual non-defeq diamond.